gene2phenotype_project/gene2phenotype_app/serializers/panel.py

```diff
@@ -5,6 +5,86 @@
 
 from ..utils import get_date_now
 
+
+class PanelCreateSerializer(serializers.ModelSerializer):
+    """
+        Panel Creation Serializer
+
+        Args:
+            serializers (_type_): 
+                name - A required char field
+                description - An optional char field
+
+
+        Raises:
+            serializers.ValidationError: Raises a validation error when the panel exists
+
+        Returns:
+            _type_: A created panel
+    """    
+    name = serializers.CharField(required=True)
+    description = serializers.CharField(required=False)
+
+    def validate(self, attrs):
+        """
+            Validate the request data 
+
+            Args:
+                attrs (_type_): A dictionary like object containing the request data
+
+            Raises:
+                serializers.ValidationError: Raises a validation error when the panel exists
+
+            Returns:
+                _type_: A validated request object
+        """        
+        name = attrs.get('name')
+        if Panel.objects.filter(name=name).exists():
+            raise serializers.ValidationError({'message': 'Can not create an existing panel'})
+
+        return attrs
+    
+    def create(self, validated_data):
+        """
+            Creation of the panel if panel does not exists
+            Updating the panel if is_visible = 0
+
+            Args:
+                validated_data (_type_): validated request object
+
+            Returns:
+                _type_: Created panel
+        """        
+        name = validated_data.get('name')
+        description = validated_data.get('description', '')
+
+        if Panel.objects.filter(name=name, is_visible=0).exists():
+            self.update(name)
```

Comment:
If the panel existing and it's `is_visible=0`, then it automatically updates to `is_visible=1`.
There are two issues:
1. the validation fails if the panel already exists so the code never gets here
2. we don't want to automatically change `is_visible` when creating a panel, updating `is_visible` should be done in a separate endpoint and `is_visible` has to be passed as an argument.

Reply:
I updated the code and tried to account for all ways, please let me know if there is something i still missed